### ISSUE TITLE
Suppress SA1010 and SA1012 for list patterns until rules are decided

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/StyleCopAnalyzers#3638]
```cs
if (e is { RemovedItems: [{ } removed] })
{
    ...
}
```

Currently the above needs #pragma which adds noise to the source.
My suggestion is to disable the SA1010 and SA1012 warnings fr list patterns until expected format is agreed on.

**Comment by jnm2** [comment]
Does this boil down to a duplicate of https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3503 and https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3509?
